src/cata_imgui.cpp

```diff
@@ -365,27 +365,46 @@ static void AddGlyphRangesMisc( UNUSED ImFontGlyphRangesBuilder *b )
     b->AddRanges( &superscripts[0] );
 }
 
-static void load_font( ImGuiIO &io, const std::vector<std::string> &typefaces,
+
+// Load the first font that exists in typefaces, falling back to unifont
+// if none of them exist.
+// - typefaces is a list of paths.
+static void load_font( ImGuiIO &io, const std::vector<font_config> &typefaces,
                        const ImWchar *ranges )
 {
-    std::vector<std::string> io_typefaces{ typefaces };
+    std::vector<font_config> io_typefaces{ typefaces };
     ensure_unifont_loaded( io_typefaces );
 
-    auto it = std::find_if( io_typefaces.begin(),
-                            io_typefaces.end(),
-    []( const std::string & io_typeface ) {
-        return file_exist( io_typeface );
-    } );
-    std::string existing_typeface = *it;
+    auto it = std::begin( io_typefaces );
+    for( ; it != std::end( io_typefaces ); ++it ) {
+        if( !file_exist( it->first ) ) {
+            debugmsg( "Font file '%s' does not exist.", it->first );
+        }
+        break;
+    }
+    if( it == std::end( io_typefaces ) ) {
+        debugmsg( "No fonts were found in the fontdata file." );
+    }
+
     ImFontConfig config = ImFontConfig();
-#ifdef IMGUI_ENABLE_FREETYPE
-    if( existing_typeface.find( "Terminus.ttf" ) != std::string::npos ||
-        existing_typeface.find( "unifont.ttf" ) != std::string::npos ) {
-        config.FontBuilderFlags = ImGuiFreeTypeBuilderFlags_ForceAutoHint;
+    switch( it->second ) {
+        case FontHint::Auto:
+            config.FontBuilderFlags = ImGuiFreeTypeBuilderFlags_ForceAutoHint;
+            break;
+        case FontHint::Bitmap:
+            config.FontBuilderFlags = ImGuiFreeTypeBuilderFlags_Bitmap;
```

Comment:
"bitmap" is not a type of hinting at all. When specified, it makes the freetype renderer ignore any vector glyphs in the font and use the bitmap glyphs instead.

Should we have a `type` field separate from `hinting` that specifies vector vs bitmap?

I would also like to have an `antialiasing` field that when set to "off" sets both `ImGuiFreeTypeBuilderFlags_Monochrome` and `ImGuiFreeTypeBuilderFlags_MonoHinting`.

Reply:
> Should we have a `type` field separate from `hinting` that specifies vector vs bitmap?

Setting the flag to bitmap means any other hinting options will be ignored, so from a usability perspective I think it's better for it to be an option in hinting even if it isn't technically correct.

> I would also like to have an `antialiasing` field that when set to "off" sets both `ImGuiFreeTypeBuilderFlags_Monochrome` and `ImGuiFreeTypeBuilderFlags_MonoHinting`.

Done.